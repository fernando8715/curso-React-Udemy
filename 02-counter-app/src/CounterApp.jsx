import { useState } from 'react';
import PropTypes from 'prop-types';


export const CounterApp = ( {value} )=> {

    const [counter, setCounter] = useState( value );
    
    const increm = ()=> {
        setCounter( counter + 1);
    }

    const decrem = ()=> {
        setCounter( counter -1);
    }

    const reset = ()=> {
        setCounter( value);
    }

    return (
        <>
            <h1>CounterApp</h1>
            <h2> {counter} </h2>
            <button aria-label='btn-add' onClick={ increm }>+1</button>
            <button onClick={ decrem }>-1</button>
            <button aria-label='btn-reset' onClick={reset}>Reset</button>
        </>
    )
} 

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
}