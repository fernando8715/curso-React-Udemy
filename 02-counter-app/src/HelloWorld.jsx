export const Hello = ()=> {
    return <h1>Hola Fernando segundo componente</h1>
}
