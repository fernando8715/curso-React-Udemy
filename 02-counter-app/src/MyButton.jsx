
export const MyButton = ({nombre, fn})=> {
    return (
        <button onClick={fn}> {nombre} </button>
    )
}
