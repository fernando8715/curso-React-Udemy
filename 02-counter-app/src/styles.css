html, body {
    background-color: #23212A;
    color: white;
    font-family: Helvetica, Arial, sans-serif;
    padding: 20px;

}
h1 {
    font-size: 30px;
}

h2 {
    padding-left: 50px;
}

button {
    background-color: white;
    color: black;
    padding: 5px;
    margin-right: 10px;
}