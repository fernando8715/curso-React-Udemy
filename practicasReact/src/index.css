:root {
  --fuentePrincipal: 'Roboto', sans-serif;
  --fuenteSecundaria: 'Lato', sans-serif;

  --primario: #1073BA;
  --secundario: #00DA55;
  --negro: #1F1E1E;
  --gris: #3B3B3B;
  --grisClaro: #e1e1e1;
  --blanco: #FFF;
  --separacion: 5rem;
}

/* Etiquetas Globales */

html {
  box-sizing: border-box;
}

*, *before, *after {
  box-sizing: inherit;
  font-size: 62.5%;
}


body {
  font-family: var(--fuentePrincipal);
  font-size: 1.6rem;
  line-height: 1.5rem;
}

h1, h2, h3 {
  font-family: var(--fuentePrincipal);
  font-weight: 900;
  margin: calc(var(--separacion)/2) 0;
}

h1 {
  font-family: var(--fuenteSecundaria); 
  font-size: 4.4rem;
}

h2 {
  font-size: 3.6rem;
}

h3 {
  font-size: 2.8rem;
}

img {
  max-width: 100%;
  display: block;
  height: auto;
}

a {
  text-decoration: none;
}

/* Clases globales */

.contenedor {
  max-width: 120rem;
  width: 90%;
  margin: 0 auto;
}


/* Utilidades */
.text-center {
  text-align: center;
}