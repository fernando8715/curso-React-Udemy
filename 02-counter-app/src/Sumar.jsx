import {useState} from 'react';
import { MyButton } from './MyButton';

export const Sumar = ()=> {

    const [counter, setCounter] = useState(0);

    const Add = ()=> {
        setCounter(counter + 5);
    }
    const Rest = ()=> {
        setCounter(counter - 1);
    }
    const Reset = ()=> {
        setCounter(0);
    }

    return (
        <>
            <h2>Funcion de suma</h2>
            <h2> {counter} </h2>
            <MyButton nombre={'Sumar'} fn={Add}/>
            <MyButton nombre={'Restar'} fn={Rest}/>
            <MyButton nombre={'Limpiar'} fn={Reset}/>
        </>
    )
}