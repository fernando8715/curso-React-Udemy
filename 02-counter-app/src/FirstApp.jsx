import PropTypes from 'prop-types';
import { Sumar } from './Sumar';

export const App = ( {title, subtitle, name} )=> {
    return (
        <>
            <h1>{ title }</h1>
            <p>{ subtitle }</p>
            <p> {name} </p>
            <Sumar/>
        </>
    )
}

//-----------PROP-TYPES-----------------
// usualmente la configuracion de los componentes se colocan al final, fuera del componente
// ejecurar el comando  yarn add prop-types
// para agregar el paquete Proptypes y poder importarlo en el proyecto
// un valor numerico en una propiedad se hace usando corchetes { }

App.propTypes = {
    title: PropTypes.string.isRequired, // para tipar una propiedad
    subtitle: PropTypes.string,
}

// default Props
// Los default props se ejecutan antes de los proptypes
// en los default props se pueden colocar valores por defecto

App.defaultProps = {
    title: 'Hola Fernando',
    subtitle: 'No hay subtitulo',
    name: 'Fernando Ruiz'
}
