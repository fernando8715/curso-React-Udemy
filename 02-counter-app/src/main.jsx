import React from 'react'
import ReactDOM from 'react-dom/client';  //
import { App } from './FirstApp';
// import { Hello } from './HelloWorld';
import { CounterApp} from './CounterApp';
// import { Sumar } from './Sumar';

import './styles.css';


ReactDOM.createRoot(document.getElementById('root') ).render(
    <React.StrictMode>
        <CounterApp value = {100} />
        {/* <Sumar value = {0} /> */}
        {/* <App title={'Aprendiendo compontes'} subtitle={'Agregando componentes'}/> */}
    </React.StrictMode>
)