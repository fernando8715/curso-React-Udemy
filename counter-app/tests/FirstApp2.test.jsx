import { render, screen } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"

describe('pruebas en FirstApp', () => {
  const title = 'Soy Fernando';
  const subtitle = 'Ingeniero informatico y desarrollador web';

  test('should match with snapshot', () => {

    const { container } = render(<FirstApp title={title} />)
    expect(container).toMatchSnapshot();
  })
  
  test('should show message "Soy Fernando"', () => {
    
    render(<FirstApp title={title} />);
    expect( screen.getByText(title) ).toBeTruthy();
  });

  test('show title in h1', () => {
    render(<FirstApp title={title}/>);
    expect(screen.getByRole('heading', {level: 1}).innerHTML).toContain(title);
  });

  test('show subtitle send by props', () => {
    
    render(<FirstApp title={title} subTitle={subtitle}/>);
    expect(screen.getAllByText(subtitle).length).toBe(2);
        
  });

})