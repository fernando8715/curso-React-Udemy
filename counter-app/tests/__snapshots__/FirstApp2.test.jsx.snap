// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en FirstApp should match with snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Soy Fernando
  </h1>
  <p>
    No hay subtítulo
  </p>
  <p>
    No hay subtítulo
  </p>
  <p>
    Fernando Herrera
  </p>
</div>
`;
